authapi/tests/test_users.py

```diff
@@ -108,17 +109,49 @@ def test_create_superuser(self):
         }
         response = self.client.post(reverse('user-list'), data=data)
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
-        [user] = User.objects.all()
+        [user] = User.objects.filter(pk=response.data['id'])
         self.assertEqual(user.username, data['email'])
         self.assertEqual(user.first_name, data['first_name'])
         self.assertEqual(user.last_name, data['last_name'])
         self.assertEqual(user.is_superuser, data['admin'])
         self.assertTrue(check_password(data['password'], user.password))
 
+    def test_permission_create_superuser(self):
+        '''Only admin users should be able to create other admin users.'''
+        data = {
+            'email': 'user1@example.org',
+            'password': 'testpassword',
+            'first_name': 'user1',
+            'last_name': 'example',
+            'admin': True,
+        }
+        resp = self.client.post(reverse('user-list'), data=data)
+        self.assertEqual(resp.status_code, status.HTTP_401_UNAUTHORIZED)
+
+        user, token = self.create_user()
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+        resp = self.client.post(reverse('user-list'), data=data)
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
+
+        self.add_permission(user, 'org:admin')
+        resp = self.client.post(reverse('user-list'), data=data)
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
+
+        self.add_permission(user, 'user:create')
+        resp = self.client.post(reverse('user-list'), data=data)
+        self.assertEqual(resp.status_code, status.HTTP_403_FORBIDDEN)
+
+        user, token = self.create_admin_user()
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+        resp = self.client.post(reverse('user-list'), data=data)
+        self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
```

Comment:
Fixed in https://github.com/praekelt/seed-auth-api/pull/27/commits/c6884606faa544171404fb12b58b8993cd487e48